compliance/notify.py

```diff
@@ -1018,30 +1018,44 @@ def _get_alerts(self, accreditation):
         conf = self._config[accreditation]
         pd_service = conf['service_id'] if isinstance(conf, dict) else conf
 
-        # Get all triggered alerts
-        # NOTE: statuses[]=triggered is supposed to work, but doesn't
-        # so we have to filter the response. leaving it in the query,
-        # as it will be a bit more efficient if they fix the issue.
-        pages = pagerduty.get(
-            'alerts',
+        # Get all current incidents for the service, we need to specify both
+        # acknowledged and triggered so we do not send a second page if there
+        # has already been one acknowledged
+        incidents_data = pagerduty.get(
+            'incidents',
             params={
                 'service_ids[]': pd_service,
-                'statuses[]': 'triggered',
+                'statuses[]': ['acknowledged', 'triggered'],
                 'time_zone': 'UTC'
             },
             creds=self._creds
         )
-        alerts = [
-            {
-                'test': alert['alert_key'],
-                'details': alert.get('body', {}).get('details', ''),
-                'created_at': alert['created_at']
-            }
-            for page in pages
-            for alert in page.json().get('alerts', [])
-            if alert['status'] == 'triggered'
-        ]
-
+        alerts = []
+
+        # If there are any incidents then loop through them and fetch all the
+        # alerts that are related to the passed accreditation
+        for inc_data in incidents_data:
+            incidents = inc_data.json()['incidents']
```

Comment:
This will fail if `incidents` is not part of the message, which it wasn't happening before thanks to `.get('alerts', [])`

Reply:
When passed a Service ID that does not exist or one that does not have a page then `incidents` is always included.
```
curl "https://api.pagerduty.com/incidents?service_ids%5B%5D=PXXXXXX" -H "Authorization: Token token=XXX" -H "Accept: application/vnd.pagerduty+json;version=2" -s | jq .
{
  "incidents": [],
  "limit": 25,
  "offset": 0,
  "total": null,
  "more": false
}
```

This however is not the case when we pass one that does not start with `P`. 
```
$ curl "https://api.pagerduty.com/incidents?service_ids%5B%5D=XXXXXX" -H "Authorization: Token token=XXX" -H "Accept: application/vnd.pagerduty+json;version=2" -s | jq .
{
  "error": {
    "message": "Invalid Input Provided",
    "code": 2001,
    "errors": [
      "Service ids must be a valid ID."
    ]
  }
}
```

Which can be solved by adding a `raise_for_status()` call.